runtime/JavaScript/src/antlr4/misc/BitSet.js

```diff
@@ -48,6 +91,33 @@ export default class BitSet {
     }
 
     get length(){
-        return this.values().length;
+        return this.data.map(l => BitSet._bitCount(l)).reduce((s,v) => s + v, 0);
+    }
+
+    _resize(index) {
+        const count = index + 32 >>> 5;
+        if (count <= this.data.length) {
+            return;
+        }
+        const data = new Uint32Array(count);
```

Comment:
Why are you constructing a new array when I think Uint32Array.resize() is available?

Reply:
I'm afraid your thinking is incorrect. You can resize an ArrayBuffer but not a TypedArray. I suspect the reason for this is that the underlying buffer is shared (a poor design decision if you ask me) so resizing a view may actually silently resize other views. Resizing the buffer would be a hack, not something I'm comfortable with